doc/whats-new.rst

```diff
@@ -55,6 +55,12 @@ Enhancements
   (:issue:`576`).
   By `Stephan Hoyer <https://github.com/shoyer>`_.
 
+- Support using an existing, opened netCDF4 ``Dataset`` with
+  :py:class:`~xarray.backends.NetCDF4DataStore`. This permits creating an
```

Comment:
What do you think about moving the current constructor logic into a classmethod `NetCDF4DataStore.open(filename, mode, format, group, writer, clobber, diskless, persist, autoclose)`

And adjusting the constructor: `NetCDF4DataStore.__init__(self, netcdf4_dataset, opener=None, writer=None, autoclose=None)`.

Right now, I don't think anyone is using the `NetCDF4DataStore` constructor directly -- there's literally no good reason for that. This also gives us a pattern we could use for other constructors (e.g., pydap) where passing in an existing object is desirable.